packages/edit-post/src/components/layout/style.scss

```diff
@@ -31,24 +27,54 @@
 			z-index: 1;
 		}
 	}
+}
 
-	& .components-resizable-box__handle-top {
-		top: 0;
-		box-shadow: 0 $border-width $gray-300;
+.edit-post-meta-boxes-main__resize-handle {
+	display: flex;
+	// The position is absolute by default inline style of ResizableBox.
+	inset: 0 0 auto 0;
+	height: 23px;
+	box-shadow: 0 $border-width $gray-300;
+
+	&:hover > button::before,
+	&:hover > button::after {
+		background-color: var(--wp-admin-theme-color);
+	}
```

Comment:
It’s a good question. I wanted to transition only compositor-friendly properties and `width` isn’t one. It’s true a single pseudo-element could use `transform: scaleX()` but then the `border-radius` would be deformed slightly (yet likely not noticeably to most folks). Does that explain it well enough and seem justifiable? I’d be glad to add a comment since the choices certainly aren’t self-explanatory.

Reply:
Scratch my prior comment. 9fd8e14f03ee95c5b3a8148267c4f5f69aac3982 is a better idea on how to avoid deforming the border radius. `clip-path` FTW and I’ve read that at least in [some browsers it can be composited](https://motion.dev/docs/performance#use-alternative-styles) too.

Reply:
The clip-path approach works fine, but in my environment, the rounded corners look slightly blurry. Can't we simply use the width+border-radius approach?

<details><summary>Diff example</summary>

```diff
diff --git a/packages/edit-post/src/components/layout/style.scss b/packages/edit-post/src/components/layout/style.scss
index f3fff8bef0..190de2989c 100644
--- a/packages/edit-post/src/components/layout/style.scss
+++ b/packages/edit-post/src/components/layout/style.scss
@@ -44,18 +44,17 @@
                border: none;
                outline: none;
                background-color: $gray-300;
-               // The visible width is the first unit, the rest is clipped when not hovered/focused.
-               width: $grid-unit-80 + $grid-unit-10 * 2;
+               width: $grid-unit-80;
                height: $grid-unit-05;
-               clip-path: inset(0 $grid-unit-10 round $grid-unit-05);
-               transition: clip-path 0.3s ease-out;
+               border-radius: $grid-unit-05;
+               transition: width 0.3s ease-out;
                @include reduce-motion("transition");
        }
 
        &:hover > button,
        > button:focus {
                background-color: var(--wp-admin-theme-color);
-               clip-path: inset(0 0 round $grid-unit-05);
+               width: $grid-unit-80 + $grid-unit-20;
        }
 }
```

</details> 

The screenshot below shows what it looks like when I change the zoom level to 150% in Chrome browser. The clip-path approach (left column) appears to have a slight loss of rounded corners.

| Before | After |
|--------|--------|
| ![before-default](https://github.com/user-attachments/assets/9a3aca30-7878-4b76-b81d-6ba4f2dd2d47)| ![default-after](https://github.com/user-attachments/assets/2897d529-3c10-47ce-a3c1-094fb627510e)| 
| ![hover-befire](https://github.com/user-attachments/assets/2434040d-9417-4445-82f9-bb19868006ed)| ![hover-after](https://github.com/user-attachments/assets/5e93a5f6-15eb-44fc-b48b-c7d1ccb0cd10)|

Reply:
I’ve done as you suggested because I can’t really make the argument that `clip-path` will perform any better. Though I do think the rounder corner variability you’d noted maybe could have been fixed. The radius applied in the clip-path was 4px when it really only needed to be 2px.